Cpp/odin-views/app.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+    Copyright 2018 Felspar Co Ltd. <http://odin.felspar.com/>
+
+    Distributed under the Boost Software License, Version 1.0.
+    See <http://www.boost.org/LICENSE_1_0.txt>
+*/
+
+#include <odin/app.hpp>
+#include <odin/credentials.hpp>
+#include <odin/odin.hpp>
+#include <odin/user.hpp>
+#include <odin/views.hpp>
+
+#include <fost/exception/parse_error.hpp>
+#include <fost/insert>
+#include <fost/log>
+#include <fostgres/sql.hpp>
+
+
+namespace {
+
+    const fostlib::module c_odin_app(odin::c_odin, "app.cpp");
+
+    fostlib::json parse_payload(fostlib::http::server::request &req) {
+        // TODO: Support multiple ContentType
+        auto body_str = fostlib::coerce<fostlib::string>(
+            fostlib::coerce<fostlib::utf8_string>(req.data()->data()));
+        return fostlib::json::parse(body_str);
+    }
+
+    const class app_login : public fostlib::urlhandler::view {
+    public:
+        app_login()
+        : view("odin.app.login") {
+        }
+
+        std::pair<boost::shared_ptr<fostlib::mime>, int> operator () (
+            const fostlib::json &config,
+            const fostlib::string &path,
+            fostlib::http::server::request &req,
+            const fostlib::host &host
+        ) const {
+            const auto paths = fostlib::split(path, "/");
+            if ( paths.size() != 1)
+                throw fostlib::exceptions::not_implemented(__func__,
+                    "Must pass app_id in the URL");
+            fostlib::pg::connection cnx{fostgres::connection(config, req)};
+            const auto app_id = paths[0];
+            fostlib::json app = odin::app::get_detail(cnx, app_id);
+            cnx.commit();
+            if ( app.isnull() )
+                throw fostlib::exceptions::not_implemented(__func__, "App not found");
```

Comment:
Should just be able to do a 404 here, can do it as view forwarding.

Reply:
I don't want user to be able to do brute force with app_id. If we return 404, attacker can use it to find app_ids in the system